src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -521,43 +803,36 @@ function AnalyticsService:LogPlayerEvent(data: PlayerEvent): { [any]: any }
 		elseif data.value ~= nil and typeof(data.value) ~= "number" then
 			return reject("AnalyticsService.LogPlayerEvent - value must be a number")
 		end
-		
-		-- Trim trailing colon
-		if string.sub(data.event, #data.event) == ":" then
-			data.event = string.sub(data.event, 1, #data.event - 1)
-		end
-		
-		GameAnalytics:addDesignEvent(data.userId, {
+
+		-- Sanitize event string to prevent errors
+		data.event = string.gsub(data.event, "[^%w%-_%.%s:]", "_")
+
+		-- Trim trailing colon and spaces
+		data.event = string.gsub(data.event, ":%s*$", "")
+
+		self:_wrapper("addDesignEvent", data.userId, {
 			eventId = data.event,
-			value = data.value
+			value = data.value,
 		})
-		
-		return resolve()
+
+		resolve()
 	end)
 end
 
 --[=[
-	This function should be called when a successful marketplace purchase is made
-	such as a gamepass or developer product
-	
-	Set `logDevProductPurchases` to false when configuring AnalyticsService if you prefer to log
-	developer product purchases within MarketplaceService.ProcessReceipt
-	
-	```lua
-	-- Inside MarketplaceService.ProcessReceipt
-	-- before returning Enum.ProductPurchaseDecision.PurchaseGranted
 	AnalyticsService:LogMarketplacePurchase({
 		userId = player.UserId,
 		itemType = "Product",
-		id = 000000000, -- Developer product id
+		id = 000000000, -- Asset Id
 		cartType = "PromptPurchase",
+		robuxPrice = 100
 	})
 	```
-	
+
 	@param data MarketplacePurchaseEvent
-	@return { [any]: any }
+	@return Promise
```

Comment:
```suggestion
	@return Promise<T> 
```

Unless the type is already defined, you need to define custom return types with Moonwave by denoting them with a <T> after the type (Eg. `Promise<T>`)